service/src/main/java/bio/terra/profile/app/common/MetricUtils.java

```diff
@@ -0,0 +1,43 @@
+package bio.terra.profile.app.common;
+
+import bio.terra.profile.model.CloudPlatform;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import io.micrometer.core.instrument.Metrics;
+import java.util.concurrent.Callable;
+
+public class MetricUtils {
+
+  private static final String CLOUD_PLATFORM_TAG = "cloudPlatform";
+
+  /**
+   * Emit a metric for the duration of creating a billing profile.
+   *
+   * @param callable the code to execute to create the profile, which will be timed
+   * @param platform the platform the profile will be created on
+   */
+  public static BillingProfile recordProfileCreation(
+      Callable<BillingProfile> callable, CloudPlatform platform) {
+    try {
+      return Metrics.globalRegistry
+          .timer("profile.creation.time", CLOUD_PLATFORM_TAG, platform.toString())
+          .recordCallable(callable);
+    } catch (RuntimeException ex) {
```

Comment:
This is ugly, but the recordCallable method declares that it throws Exception (to bubble up anything from the callable). But the actual method we are calling (submitAndWait) does not declare any checked exceptions.

Member
Author
@[cahrens](https://github.com/cahrens) cahrens [16 hours ago](https://github.com/DataBiosphere/terra-billing-profile-manager/pull/101#discussion_r1043840065)
(And I had to switch to using a Callable instead of a Runnable to get the return value. Interestingly, the record API with a Runnable does not declare that it throws Exception)

Reply:
I would rather stick with this code, as it seems simpler (the ugliness is hidden in the util method instead of the caller having to keep to handle start/stop or determine the elapsed time).